evmlab/reproduce.py

```diff
@@ -149,15 +156,15 @@ def reproduceTx(txhash, vm, api):
         done = True
         # Add accounts that we know of 
         for addr in list(externals_tofetch):
-            acc = api.getAccountInfo( addr , blnum)
+            acc = api.getAccountInfo( addr , blnum - 1)  # need to load accountInfo at block before tx
```

Comment:
I think this is correct as we want the genesis block to contain the state right before the tx occurred.

This puts the correct nonce for the accounts, thus reproducing the correct contract addresses for and CREATE opcode.

One case where I think this would not work is if there are 2 txs in the same block for any account touched by the tx we are reproducing.

Reply:
Nice catch!